synapseclient/api/entity_services.py

```diff
@@ -542,3 +543,327 @@ async def get_entities_by_md5(
     return await client.rest_get_async(
         uri=f"/entity/md5/{md5}",
     )
+
+
+async def get_entity_provenance(
+    entity_id: str,
+    version_number: Optional[int] = None,
+    *,
+    synapse_client: Optional["Synapse"] = None,
+) -> Dict[str, Any]:
+    """
```

Comment:
Should the dict this returns be a dataclass?

Reply:
Not in this case. The responsibility for these API modules has been to "talk" in the same language as the Rest API. So in this case it's to accept and return back the dictionaries that the API would accept/return. Then - Where this is being used within the Dataclass (Activity) is where it is usually ported into a Dataclass.